codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/protocol/ServerProtocolTestGenerator.kt

```diff
@@ -244,27 +241,35 @@ class ServerProtocolTestGenerator(
         writeInline("let output =")
         instantiator.render(this, expectedShape, testCase.params)
         write(";")
+        val operationName = if (expectedShape.hasTrait(ErrorTrait::class.java)) {
+            "${operationSymbol.name}${ServerHttpProtocolGenerator.OPERATION_ERROR_WRAPPER_SUFFIX}"
+        } else {
+            "${operationSymbol.name}${ServerHttpProtocolGenerator.OPERATION_OUTPUT_WRAPPER_SUFFIX}"
+        }
         rustTemplate(
             """
-            use #{SerializeHttpResponse};
-            let op = #{op}::new();
-            let http_response = op.serialize(&output).expect("unable to serialize `#{op}` into HTTP response body");
+            let output = super::$operationName(output);
+            use #{Axum}::response::IntoResponse;
+            let http_response = output.into_response();
             """,
             *codegenScope,
-            "op" to operationSymbol,
         )
-        rust("""
+        rust(
+            """
             assert_eq!(
                 http::StatusCode::from_u16(${testCase.code}).expect("invalid expected HTTP status code"),
                 http_response.status()
             );
-        """)
+            """
+        )
         if (testCase.body != null) {
-            rust("""
-                let body = std::str::from_utf8(http_response.body())
-                    .expect("serialized response body does not contain valid UTF-8");
+            rustTemplate(
+                """
+                let body = #{Hyper}::body::to_bytes(http_response.into_body()).await.expect("unable to extract body to bytes");
```

Comment:
Instead of directly using Hyper body, let's use our reexport https://github.com/awslabs/smithy-rs/pull/839/files#diff-3f18776c19e135735316c67a5284d096662a25ba8f2d1677c1e36ef50f46e6a6R44

Reply:
Yep, good idea!

Reply:
It's ok to use `Hyper` here to convert to bytes, the tests can use whatever they want and we should not impose the tests to use a conversion function provided by us.